sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnosticsContext.java

```diff
@@ -347,6 +350,28 @@ public Set<String> getContactedRegionNames() {
         return regionsContacted;
     }
 
+    /**
+     * Returns the system usage
+     * NOTE: this information is not included in the json representation returned from {@link #toJson()} because it
+     * is usually only relevant when thresholds are violated, in which case the entire diagnostics json-string is
+     * included. Calling this method will lazily collect the user agent - which can be useful when writing
+     * a custom {@link CosmosDiagnosticsHandler}
+     * @return the system usage
+     */
+    public Collection<String> getSystemUsage() {
+        TreeSet<String> systemUsage = new TreeSet<>();
```

Comment:
Why do we need a `TreeSet` here?

Reply:
Ordered and unique - different CosmosDiagnostics instances might have values spanning different time windows. I want to exclude duplicates - but span the entire time period.